transport.go

```diff
@@ -317,7 +317,7 @@ func (p *connPool) ref() {
 }
 
 func (p *connPool) unref() {
-	if atomic.AddUintptr(&p.refc, ^uintptr(0)) == 0 {
+	if atomic.AddUintptr(&p.refc, ^uintptr(0)) != 0 {
```

Comment:
The reference counter here intends to track the number of inflight operations so we delay closing the pool until the last inflight operations has completed.

At first sight it seems like this change would result in closing the connection pool on the first call to `unref` isn't it?